network-config-analyzer/NetworkConfigQuery.py

```diff
@@ -531,39 +532,104 @@ class ConnectivityMapQuery(NetworkConfigQuery):
     def get_supported_output_formats():
         return {'txt', 'yaml', 'csv', 'md', 'dot'}
 
+    def is_subset_active(self):
+        """
+        returns indication if the subset feature was activated
+        :return: True/False
+        """
+        if self.output_config.subset:
+            return True
+        return False
+
+    def is_in_subset(self, peer):
+        """
+        returns indication if the peer element is in the defined subset
+        Please note: Subset is a sort of a filter. It filters out elements which WERE DEFINED and do not match the settings.
+        Thus, the function returns True if no subset was defined
+        since, in this case, the subset is infinite and everything is "in the subset"
+        :param peer: peer element to filter (currently supports only Pods)
+        :return:
+        """
+        # if subset restrictions were not defined at all, everything is in the subset
+        if not self.is_subset_active():
+            return True
+
+        subset = self.output_config.subset
+        # filter by namespace
+        if isinstance(peer, Peer) and subset.get('namespace_subset') \
+                and str(peer.namespace) in str(subset['namespace_subset']):
+            return True
+
+        # filter by deployment
+        if isinstance(peer, Pod) and subset.get('deployment_subset') and peer.owner_name:
+            dep_names = str(subset.get('deployment_subset')).split(',')
```

Comment:
why is splitting for labels done in `nca.py`, while for deployments it is done here? Maybe `OutputConfiguration.py` is the right place for both these preprocessing?

Reply:
In nca.py we need to 'normalize' the format given by the CLI so it will be consistent with the one made by the yaml parser. In the labels case it is splitting and dict-ing.
In NetworkConfigQuery.py the code pre-process the data (deployments, labels and namespaces) so it can use it for its logic.